### PR TITLE
Add a BBE for expression-bodied functions

### PR DESCRIPTION
**Pull request by pubudu91**

### CODE REVIEW

File: examples/expression-bodied-functions/expression_bodied_functions.bal [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+// of the function.
+function add(int a, int b) returns int => a + b;
+
+public function main() {
+    // There is no difference in how you call an expression-bodied function.
+    int sum = add(10, 20);
+    io:println(sum);
+
+    // Anonymous functions can have expression bodies as well.
+    // Here we use an expression bodied anonymous function to easily map a
```
Comment (praneesha) [inline examples/expression-bodied-functions/expression_bodied_functions.bal]:
```suggestion
    // Here, an expression bodied anonymous function is used to easily map a
```

File: examples/expression-bodied-functions/expression_bodied_functions.bal [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import ballerina/io;
+
+// Note the syntactic difference in the function body from a regular block function body.
```
Comment (praneesha) [inline examples/expression-bodied-functions/expression_bodied_functions.bal]:
```suggestion
// Note the syntactic difference in the function body of an expression-bodied function compared to a regular block function body.
```

File: examples/expression-bodied-functions/expression_bodied_functions.out [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# To run this sample, navigate to the directory that contains the
+# `.bal` file, and execute the `ballerina run` command.
```
Comment (praneesha) [inline examples/expression-bodied-functions/expression_bodied_functions.out]:
```suggestion
# `.bal` file and execute the `ballerina run` command.
```

File: examples/expression-bodied-functions/expression_bodied_functions.description [hunk 4]
```diff
@@ -0,0 +1,2 @@
+// A function body can be a single expression. Such a function is called an expression-bodied function and the
```
Comment (hasithaa) [inline examples/expression-bodied-functions/expression_bodied_functions.description]: We haven't used the term `expression-bodied function` in the spec. Shall we avoid this term?

Comment (pubudu91) [reply examples/expression-bodied-functions/expression_bodied_functions.description]: Picked up the term from https://github.com/ballerina-platform/ballerina-spec/issues/395. If not, we don't have a simple enough term to refer to it right?

**Pull request merged**